inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/sidebar/ExternalSearchAnnotationSidebar.java

```diff
@@ -0,0 +1,421 @@
+import org.apache.wicket.extensions.markup.html.repeater.data.table.DataTable;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.DefaultDataTable;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.IColumn;
+import org.apache.wicket.markup.html.WebMarkupContainer;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.form.ChoiceRenderer;
+import org.apache.wicket.markup.html.form.DropDownChoice;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.html.form.TextField;
+import org.apache.wicket.markup.html.panel.Panel;
+import org.apache.wicket.markup.repeater.Item;
+import org.apache.wicket.model.*;
+import org.apache.wicket.spring.injection.annot.SpringBean;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.safety.Cleaner;
+import org.jsoup.safety.Whitelist;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wicketstuff.event.annotation.OnEvent;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import static de.tudarmstadt.ukp.inception.app.ui.externalsearch.sidebar.ExternalSearchUserStateMetaData.CURRENT_ES_USER_STATE;
+import static org.apache.commons.lang3.StringUtils.defaultIfBlank;
+
+public class ExternalSearchAnnotationSidebar
+    extends AnnotationSidebar_ImplBase
+{
+    private static final long serialVersionUID = -3358207848681467994L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(ExternalSearchAnnotationSidebar.class);
+
+    private @SpringBean DocumentService documentService;
+    private @SpringBean AnnotationSchemaService annotationService;
+    private @SpringBean ProjectService projectService;
+    private @SpringBean ExternalSearchService externalSearchService;
+    private @SpringBean UserDao userRepository;
+    private @SpringBean ImportExportService importExportService;
+    private @SpringBean ApplicationEventPublisherHolder applicationEventPublisher;
+
+    private CompoundPropertyModel<ExternalSearchUserState> searchStateModel;
+
+    final WebMarkupContainer mainContainer;
+
+    private List<ExternalSearchResult> results = new ArrayList<ExternalSearchResult>();
+
+    private IModel<List<DocumentRepository>> repositoriesModel;
+
+    private ExternalSearchResult selectedResult;
+
+    private Project project;
+
+    private WebMarkupContainer dataTableContainer;
+
+    private DocumentImporter documentImporter;
+
+
+    public ExternalSearchAnnotationSidebar(String aId, IModel<AnnotatorState> aModel,
+            AnnotationActionHandler aActionHandler, JCasProvider aJCasProvider,
+            AnnotationPage aAnnotationPage)
+    {
+        super(aId, aModel, aActionHandler, aJCasProvider, aAnnotationPage);
+
+        // Attach search state to annotation page
+        searchStateModel = new CompoundPropertyModel<>(LambdaModelAdapter.of(
```

Comment:
The sidebar persistence stuff won't be obvious to a casual reader of this code. Can you comment clearly what the relevant code does and what its motivation is?